lib/finders/StoreFinder.js

```diff
@@ -101,29 +108,62 @@ class StoreFinder extends events.EventEmitter {
 	 * Watches the components for changing.
 	 */
 	watch() {
-		if (this._fileWatcher) {
-			return;
+		if (this._readyPromise) {
+			return this._readyPromise;
 		}
 
-		this._fileWatcher = chokidar.watch(
+		const fileWatcher = chokidar.watch(
```

Comment:
We do, when a store is removed it's used to unwatch and when a store appears it's used to start watching.